src/lib/y2partitioner/widgets/btrfs_subvolumes_add_button.rb

```diff
@@ -43,33 +42,10 @@ def filesystem
         table.filesystem
       end
 
-      # Creates a new subvolume with the form values
-      # @see Dialogs::BtrfsSubvolume::Form
-      #
-      # The mount point is generated from the subvolume path and
-      # the filesystem mount point.
-      # @see #mountpoint
-      def add_subvolume(form)
-        parent = filesystem.default_btrfs_subvolume
-        subvol = parent.create_btrfs_subvolume(form.path)
-        subvol.nocow = form.nocow
-        subvol.mountpoint = mountpoint(form.path)
-      end
-
-      # Generates the subvolume mount point
-      #
-      # The subvolume mount point is generated from the subvolume path and
-      # the filesystem mount point.
-      def mountpoint(path)
-        File.join(filesystem.mountpoint, subvolume_relative_path(path))
-      end
-
-      def subvolume_relative_path(path)
-        path.sub(btrfs_root_path, "")
-      end
-
-      def btrfs_root_path
-        filesystem.default_btrfs_subvolume.path + "/"
+      def add_subvolume(path, nocow)
+        subvolume = filesystem.create_btrfs_subvolume(path, nocow)
+        # Manually created subvolumes cannot be shadowed
+        subvolume.can_be_shadowed = false
```

Comment:
Maybe this should be some kind of default, I don't see the point in setting it manually after every subvolume creation.